src/libcollections/btree/node.rs

```diff
@@ -694,15 +629,16 @@ impl<'a, K, V> NodeRef<marker::Mut<'a>, K, V, marker::LeafOrInternal> {
             let edge = match self.reborrow_mut().force() {
                 ForceResult::Leaf(_) => None,
                 ForceResult::Internal(mut internal) => {
-                    let edge = slice_remove(
-                        slice::from_raw_parts_mut(
-                            internal.as_internal_mut().edges.as_mut_ptr(),
-                            old_len+1
-                        ),
-                        0
-                    );
-
-                    let mut new_root = Root { node: edge, height: internal.height - 1 };
+                    let edge = slice_remove(slice::from_raw_parts_mut(internal.as_internal_mut()
+                                                                              .edges
+                                                                              .as_mut_ptr(),
+                                                                      old_len + 1),
+                                            0);
```

Comment:
I can see how this is more of a borderline and debatable case, but this looks significantly worse to me.